stdlib/universe/holt_winters/holt_winters.go

```diff
@@ -123,47 +99,55 @@ func (r *HoltWinters) Do(vs *array.Float64) *array.Float64 {
 	if r.seasonal {
 		size += m
 	}
-	r.tracker.track(size)
-	defer r.tracker.free(size)
-	params := make([]float64, 6, size)
-	params[4] = l0
-	params[5] = b0
+	params := mutable.NewFloat64Array(r.alloc)
+	defer params.Release()
+	params.Resize(size)
+	params.Set(4, l0)
+	params.Set(5, b0)
 	if r.seasonal {
 		for i := 0; i < m; i++ {
 			if vs.IsValid(i) {
-				params = append(params, vs.Value(i)/l0)
+				params.Set(i+6, vs.Value(i)/l0)
 			} else {
-				params = append(params, 0)
+				params.Set(i+6, 0)
 			}
 		}
 	}
 
 	// Determine best fit for the various parameters
 	minSSE := math.Inf(1)
-	var bestParams []float64
+	var bestParams *mutable.Float64Array
 	for alpha := hwGuessLower; alpha < hwGuessUpper; alpha += hwGuessStep {
 		for beta := hwGuessLower; beta < hwGuessUpper; beta += hwGuessStep {
 			for gamma := hwGuessLower; gamma < hwGuessUpper; gamma += hwGuessStep {
 				for phi := hwGuessLower; phi < hwGuessUpper; phi += hwGuessStep {
-					params[0] = alpha
-					params[1] = beta
-					params[2] = gamma
-					params[3] = phi
+					params.Set(0, alpha)
+					params.Set(1, beta)
+					params.Set(2, gamma)
+					params.Set(3, phi)
 					sse, params := r.optim.Optimize(r.sse, params, r.epsilon, 1)
 					if sse < minSSE || bestParams == nil {
+						if bestParams != nil {
+							bestParams.Release()
+						}
 						minSSE = sse
 						bestParams = params
 					}
+					if bestParams != params {
+						params.Release()
+					}
```

Comment:
`Optimize` creates new params from the given ones every time it is invoked (I'll point out in a comment where). Those params need to be released in case they are not the best ones (because we don't need them anymore.

In case that they are the new best, we can release the elder bests.